core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -377,6 +411,16 @@ private[spark] object JettyUtils extends Logging {
     ServerInfo(server, boundPort, securePort,
       server.getHandler().asInstanceOf[ContextHandlerCollection])
   }
+  /* Basic Authentication Handler */
+  private def basicAuthenticationHandler(): SecurityHandler = {
+    val csh = new ConstraintSecurityHandler();
```

Comment:
Do we need to set a new security handler for each of the ServletContextHandlers or one across them will do?
If yes, you can make it lazy so that the same instance is passed.